src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.Helper.cs

```diff
@@ -0,0 +1,136 @@
+            {
+                BuildEventString = buildEventString;
+                TargetNameString = targetNameString;
+                GetTargetString = getTargetString;
+                SetTargetDependencies = setTargetDependencies;
+            }
+
+            private Func<ProjectTargetElement, string> GetTargetString { get; }
+            private Action<ProjectTargetElement> SetTargetDependencies { get; }
+            private string BuildEventString { get; }
+            private string TargetNameString { get; }
+
+            public string GetProperty(ProjectRootElement projectXml)
+            {
+                var result = FindExecTaskInTargets(projectXml);
+
+                if (result.success == false)
+                {
+                    return null;
+                }
+
+                if (result.execTask.Parameters.TryGetValue(_commandString, out var commandText))
+                {
+                    return commandText;
+                }
+
+                return null; //unreachable
+            }
+
+            public void SetProperty(string unevaluatedPropertyValue, ProjectRootElement projectXml)
+            {
+                if (string.IsNullOrWhiteSpace(unevaluatedPropertyValue) &&
+                    !unevaluatedPropertyValue.Contains("\n"))
+                {
+                    var result = FindTargetToRemove(projectXml);
+                    if (result.success)
+                    {
+                        projectXml.RemoveChild(result.target);
+                    }
+                }
+                else
+                {
+                    SetParameter(projectXml, unevaluatedPropertyValue);
+                }
+            }
+
+            private (bool success, ProjectTaskElement execTask) FindExecTaskInTargets(ProjectRootElement projectXml)
+            {
+                var execTask = projectXml.Targets
+                                    .Where(target => GetTargetString(target) == BuildEventString)
+                                    .SelectMany(target => target.Tasks)
+                                    .Where(task => task.Name == _execTaskName)
+                                    .FirstOrDefault();
+                return (success: execTask != null, execTask: execTask);
+            }
+
+            private (bool success, ProjectTargetElement target) FindTargetToRemove(ProjectRootElement projectXml)
+            {
+                var targetArray = projectXml.Targets
+                                        .Where(target =>
+                                            GetTargetString(target) == BuildEventString &&
+                                            target.Children.Count == 1 &&
+                                            target.Tasks.Count == 1 &&
+                                            target.Tasks.First().Name == _execTaskName);
+                return (success: targetArray.Count() == 1, target: targetArray.SingleOrDefault());
```

Comment:
It does not make sense to have the same target twice but nevertheless we do allow it. If there is more than one, then we will crash since we are using `Single`